src/main/java/flaxbeard/steamcraft/tile/TileEntitySteamCharger.java

```diff
@@ -3,285 +3,347 @@
 import flaxbeard.steamcraft.api.ISteamChargable;
 import flaxbeard.steamcraft.api.ISteamTransporter;
 import flaxbeard.steamcraft.api.tile.SteamTransporterTileEntity;
+import flaxbeard.steamcraft.integration.CrossMod;
 import flaxbeard.steamcraft.item.ItemExosuitArmor;
+
 import net.minecraft.entity.item.EntityItem;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.inventory.IInventory;
+import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.network.NetworkManager;
 import net.minecraft.network.Packet;
 import net.minecraft.network.play.server.S35PacketUpdateTileEntity;
 import net.minecraftforge.common.util.ForgeDirection;
+import tconstruct.library.tools.ToolCore;
 
 public class TileEntitySteamCharger extends SteamTransporterTileEntity implements ISteamTransporter, IInventory {
 
-    public int randomDegrees;
-    private boolean isCharging = false;
-    private boolean hadItem = false;
-    private float prevPercent = 0F;
-    private ItemStack[] inventory = new ItemStack[1];
-
-    public TileEntitySteamCharger() {
-        super(new ForgeDirection[]{ForgeDirection.DOWN});
-        this.addSidesToGaugeBlacklist(new ForgeDirection[]{ForgeDirection.UP, ForgeDirection.DOWN});
-    }
-
-    @Override
-    public void readFromNBT(NBTTagCompound par1NBTTagCompound) {
-        super.readFromNBT(par1NBTTagCompound);
-        randomDegrees = (int) (Math.random() * 360);
-        if (par1NBTTagCompound.hasKey("inventory")) {
-            this.inventory[0] = ItemStack.loadItemStackFromNBT(par1NBTTagCompound.getCompoundTag("inventory"));
-        }
+  public int randomDegrees;
```

Comment:
Wrong number of spaces in the indents starting here until basically the end of the file.